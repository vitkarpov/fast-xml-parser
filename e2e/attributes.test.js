const fs = require('fs');
const path = require('path');
const {parse} = require('../dst/index');

describe('attributes', () => {
  it('parse', () => {
    const input = fs.readFileSync(path.join(__dirname, 'inputs/attributes.html'), 'utf-8');
    expect(parse(input)).toMatchSnapshot();
  });
});
