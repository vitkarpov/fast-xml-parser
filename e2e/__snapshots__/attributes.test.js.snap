// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attributes parse 1`] = `
Node {
  "attrs": Object {},
  "children": Array [
    Node {
      "attrs": Object {},
      "children": Array [
        Node {
          "attrs": Object {},
          "children": Array [
            Node {
              "attrs": Object {},
              "children": Array [],
              "name": "Attributes test",
              "type": "TEXT",
            },
          ],
          "name": "title",
          "type": "OPENING_TAG",
        },
      ],
      "name": "head",
      "type": "OPENING_TAG",
    },
    Node {
      "attrs": Object {},
      "children": Array [
        Node {
          "attrs": Object {
            "class": "value0",
            "id": "test0",
            "title": "value1",
          },
          "children": Array [
            Node {
              "attrs": Object {},
              "children": Array [],
              "name": "class=\\"value0\\" title=\\"value1\\"",
              "type": "TEXT",
            },
          ],
          "name": "button",
          "type": "OPENING_TAG",
        },
        Node {
          "attrs": Object {
            "class": "value2",
            "disabled": "true",
            "id": "test1",
          },
          "children": Array [
            Node {
              "attrs": Object {},
              "children": Array [],
              "name": "class=value2 disabled",
              "type": "TEXT",
            },
          ],
          "name": "button",
          "type": "OPENING_TAG",
        },
        Node {
          "attrs": Object {
            "class": "value4",
            "id": "test2",
            "title": "value5",
          },
          "children": Array [
            Node {
              "attrs": Object {},
              "children": Array [],
              "name": "class=\\"value4\\"title=\\"value5\\"",
              "type": "TEXT",
            },
          ],
          "name": "button",
          "type": "OPENING_TAG",
        },
      ],
      "name": "body",
      "type": "OPENING_TAG",
    },
  ],
  "name": "html",
  "type": "OPENING_TAG",
}
`;

exports[`attributes stringify 1`] = `"<html><head><title>Attributes test</title></head><body><button id=\\"test0\\" class=\\"value0\\" title=\\"value1\\">class=\\"value0\\" title=\\"value1\\"</button><button id=\\"test1\\" class=\\"value2\\" disabled=\\"true\\">class=value2 disabled</button><button id=\\"test2\\" class=\\"value4\\" title=\\"value5\\">class=\\"value4\\"title=\\"value5\\"</button></body></html>"`;
